### DISCUSSION TITLE
How do I add blocks to the block palette?

### DISCUSSION THREAD

**Discussion by FlamedDogo99** [ScratchAddons/ScratchAddons#5997] [Help]
So far I’ve only been able to find addon.tab.addBlock, to add blocks to the debug menu. I was hoping to add the counter and for loop blocks and a few others to the control and events tabs.

Thanks in advance

**Comment by Samq64** [comment]
Since Scratch Addons was never intended to have addons that addon blocks, `addon.tab.addBlock` can't do much. There's a pretty long discussion about an addon that adds blocks besides other than the debugger ones here: #817

**Comment by iqnite** [comment]
Answered at #4961
